### PR TITLE
GLES - Use TBO instead of SSBO and make shaders use it when it is not gl45

### PR DESCRIPTION
**Pull request by Cristo86**
Rollback GLES to work as GL41 does as those storage blocks are not mandatory in GLES implementations/devices.

**Test plan**
1. Launch the app on Android
2. Enter any domain or last location
3. My Avatar should render (before this PR it isn't happening).

4. Windows/Linux/macOS rendering shouldn't be affected, please check.

**Note**
On some Mali GPU Devices (Like Huawei Mate 9 Pro - Use https://play.google.com/store/apps/details?id=de.saschawillems.glescapsviewer&hl=en app to confirm, OpenGL ES - Vendor and Renderer) Interface may flicker being unusable, that's a different issue not addressed in this PR.

### CODE REVIEW

**Comment by Cristo86** [comment]
Build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13839/HighFidelity-Beta-PR14195-ec2170f.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13839/HighFidelity-Beta-PR14195-ec2170f.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13839/HighFidelity-Beta-Interface-PR14195-ec2170f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13839/HighFidelity-Beta-PR14195-ec2170f.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13839/HighFidelity-Beta-Interface-PR14195-ec2170f.exe

File: libraries/render-utils/src/Blendshape.slh [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 <@func declareBlendshape(USE_NORMAL, USE_TANGENT)@>
 
-#if defined(GPU_GL410)
+#if !defined(GPU_GL450)
```
Comment (jherico) [inline libraries/render-utils/src/Blendshape.slh]: shaders should use `GPU_SSBO_TRANSFORM_OBJECT` to test whether they should use SSBOs, not a specific GL version.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13853/HighFidelity-Beta-PR14195-837dabd.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13853/HighFidelity-Beta-PR14195-837dabd.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13853/HighFidelity-Beta-Interface-PR14195-837dabd.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13853/HighFidelity-Beta-PR14195-837dabd.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13853/HighFidelity-Beta-Interface-PR14195-837dabd.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14496/HighFidelity-Beta-PR14195-a09b1ad.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14496/HighFidelity-Beta-PR14195-a09b1ad.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14496/HighFidelity-Beta-Interface-PR14195-a09b1ad.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14496/HighFidelity-Beta-PR14195-a09b1ad.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14496/HighFidelity-Beta-Interface-PR14195-a09b1ad.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14516/HighFidelity-Beta-PR14195-bc7e494.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14516/HighFidelity-Beta-PR14195-bc7e494.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14516/HighFidelity-Beta-Interface-PR14195-bc7e494.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14516/HighFidelity-Beta-PR14195-bc7e494.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14516/HighFidelity-Beta-Interface-PR14195-bc7e494.exe

File: tools/shadergen.py [hunk 2]
```diff
@@ -172,7 +172,7 @@ def processCommand(line):
         scribeArgs = getCommonScribeArgs(scribeFile, libs)
         headerFlag = '-H'
         # using the old flag on Android builds for now
-        if (dialect == '310es'): headerFlag = '-h'
+        if (dialect == '310es'): headerFlag = '-H'
```
Comment (Cristo86) [inline tools/shadergen.py]: With this change we have a build error:
```
07:12:40   stdout:
07:12:40   /home/jenkins/workspace/pr-build/label/android/android/app/.externalNativeBuild/cmake/debug/arm64-v8a/libraries/shaders/shaders/gpu/310es/DrawTexcoordRectTransformUnitQuad.vert
07:12:40   ERROR: #version: ES shaders for Vulkan SPIR-V require version 310 or higher
07:12:40   ERROR: 1 compilation errors.  No code generated.
```
With `headerFlag = '-h'` it works but should it work with -H? @samcake @jherico

Comment (jherico) [reply tools/shadergen.py]: revert that change... that's only for building GLES on Windows platforms.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14626/HighFidelity-Beta-PR14195-fa0c58f.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14626/HighFidelity-Beta-Interface-PR14195-fa0c58f.dmg

File: tools/shadergen.py (hunk 2 above)
Comment (samcake) [reply tools/shadergen.py]: My bad, after confirming with austin, you can remove that change

File: libraries/gpu/src/gpu/ShaderConstants.h [hunk 3]
```diff
@@ -21,6 +21,12 @@
 
 #define GPU_TEXTURE_TRANSFORM_OBJECT 31
 
+
+#define GPU_RESOURCE_BUFFER_SLOT0_TEXTURE 35
+#define GPU_RESOURCE_BUFFER_SLOT1_TEXTURE 35
```
Comment (jherico) [inline libraries/gpu/src/gpu/ShaderConstants.h]: why is this the same as the previous value?

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14626/HighFidelity-Beta-PR14195-fa0c58f.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14626/HighFidelity-Beta-Interface-PR14195-fa0c58f.exe

File: libraries/gpu/src/gpu/ShaderConstants.h (hunk 3 above)
Comment (Cristo86) [reply libraries/gpu/src/gpu/ShaderConstants.h]: @samcake Is it correct, or the next one should be 36?

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14634/HighFidelity-Beta-PR14195-bcda6a7.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14636/HighFidelity-Beta-PR14195-fa0c58f.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14638/HighFidelity-Beta-PR14195-bcda6a7.apk

File: libraries/gpu/src/gpu/ShaderConstants.h (hunk 3 above)
Comment (samcake) [inline libraries/gpu/src/gpu/ShaderConstants.h]: this should be 36

Comment (samcake) [reply libraries/gpu/src/gpu/ShaderConstants.h]: it should be 36 Austin is right

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14641/HighFidelity-Beta-PR14195-c310d41.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14641/HighFidelity-Beta-PR14195-c310d41.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14641/HighFidelity-Beta-Interface-PR14195-c310d41.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14641/HighFidelity-Beta-PR14195-c310d41.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14641/HighFidelity-Beta-Interface-PR14195-c310d41.exe

**Pull request closed without merging**